### PR TITLE
Block API: Adding the keywords property to enhance filtering the blocks

### PR DESCRIPTION
**Pull request by youknowriad**
This PR adds an optional `keywords` property taken into consideration when filtering the blocks in the inserter. As an example It's used currently to display the "verse" block if we search for "poetry"

### CODE REVIEW

**Comment by jasmussen** [comment]
Love it! Works great for me. 👍 👍

File: blocks/library/verse/index.js [hunk 1]
```diff
@@ -21,6 +21,8 @@ registerBlockType( 'core/verse', {
 
 	category: 'formatting',
 
+	keywords: [ 'poetry' ],
```
Comment (mtias) [inline blocks/library/verse/index.js]: We would need to consider translation of these in the future.

**Review by mtias — APPROVED** [review]

File: blocks/library/verse/index.js (hunk 1 above)
Comment (youknowriad) [inline blocks/library/verse/index.js]: Good point. I can wrap it in `__`

Comment (mtias) [inline blocks/library/verse/index.js]: I don't see why it wouldn't.

File: blocks/api/registration.js [hunk 2]
```diff
@@ -67,6 +67,12 @@ export function registerBlockType( name, settings ) {
 		);
 		return;
 	}
+	if ( 'keywords' in settings && settings.keywords.length > 3 ) {
```
Comment (aduth) [inline blocks/api/registration.js]: What was the thinking behind the limit?

Comment (youknowriad) [inline blocks/api/registration.js]: Avoid plugin authors from abusing keywords (keep the inserter search "sane")

**Review by s-a-s-k-i-a — COMMENTED** [review]
maximum of 3 keywords allowed; however this code allows each keyword to be unlimited characters long. 

Thus this will be abused. 
We can for example have keywords defined like this:
keywords: ['wordpresspaypalwoocommercefrontendyoastsaskialundwearethegreatesbestpluginwhaterverstring', 'andwhileweareatitwecanaddsomemorestorytellinghere', 'andhereandhere'],

**Comment by mtias** [comment]
@s-a-s-k-i-a yes, you can also just do `[ 'all my keywords here', 'other keywords here' ]`

**Comment by aduth** [comment]
Honestly I'm not sure why we bother trying to restrict keywords. Seems like unnecessary hand-slapping that determined plugin developers could always get around anyways.

**Comment by mtias** [comment]
Agree, we should just remove the limitation.

**Pull request merged**